client/src/components/EmployeeTable/index.jsx

```diff
@@ -0,0 +1,223 @@
+/* eslint-disable react/prop-types */
+/* eslint-disable react/jsx-wrap-multilines */
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import axios from 'axios';
+import {
+  Box,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableRow,
+  Toolbar,
+  Typography,
+  Checkbox,
+  TablePagination,
+} from '@material-ui/core';
+import RecipientOptions from './TableFilters/TableFilters';
+
+const headCells = [
+  {
+    id: 'name',
+    numeric: false,
+    disablePadding: true,
+    label: 'Name',
+  },
+  { id: 'job-title', numeric: true, disablePadding: false, label: 'Job Title' },
+  {
+    id: 'department',
+    numeric: true,
+    disablePadding: false,
+    label: 'Department',
+  },
+  { id: 'office', numeric: true, disablePadding: false, label: 'Office' },
+  {
+    id: 'start-date',
+    numeric: true,
+    disablePadding: false,
+    label: 'Start Date',
+  },
+];
+
+const EnhancedTableHead = () => {
+  const { recipients, filteredEmployeeData } = useSelector(
+    (state) => state.employeeTableReducer,
+  );
+  const numSelected = recipients.length;
+  const dispatch = useDispatch();
+
+  const handleSelectAllClick = (event) => {
+    const checkedAndNotIndeterminate =
+      event.target.checked && !event.target.indeterminate;
+    const payload = { checked: checkedAndNotIndeterminate };
+    dispatch({ type: 'TOGGLE_FILTERED_RECIPIENTS', payload });
+  };
+
+  return (
+    <TableHead>
+      <TableRow>
+        <TableCell padding='checkbox'>
+          <Checkbox
+            indeterminate={
+              numSelected > 0 && numSelected < filteredEmployeeData.length
+            }
+            checked={numSelected === filteredEmployeeData.length}
+            onChange={handleSelectAllClick}
+            inputProps={{ 'aria-label': 'Select all employees' }}
+          />
+        </TableCell>
+        {headCells.map((headCell) => (
+          <TableCell
+            key={headCell.id}
+            align={headCell.numeric ? 'right' : 'left'}
```

Comment:
Can we use a more descriptive name for this? `numeric` doesn't seem to make sense here.